docs/rules/extensions.md

```diff
@@ -170,3 +170,10 @@ import foo from '@/foo';
 ## When Not To Use It
 
 If you are not concerned about a consistent usage of file extension.
+
+You should not configure this rule to be "never" if you are using ECMAScript modules directly in `node`.
```

Comment:
```suggestion
In the future, when this rule supports native node ESM resolution (and the plugin is configured to use native, instead of transpiled, ESM, a config option that is not yet available), setting this to `always` will have no effect.
```